controllers/storageclient/badgerdb/simple.go

```diff
@@ -48,5 +64,256 @@ func GetClient(opts badger.Options) (*Client, error) {
 	}
 	client.db = db
 
+	// add additionalOptions
+	a := getDefaultAdditionalOptions()
+	err = mergo.Merge(&a, additionalOptions)
+	if err != nil {
+		return nil, fmt.Errorf("cannot merge additionalOptions with defaultOptions for BadgerDB: %e", err)
+	}
+	client.additionalOptions = a
+
 	return &client, nil
 }
+
+func getDefaultAdditionalOptions() AdditionalOptions {
+	return AdditionalOptions{
+		TTL: time.Hour * 24,
+	}
+}
+
+func getValueFromBadgerDB(db *badger.DB, key string) ([]byte, error) {
+	var valCopy []byte
+
+	err := db.View(func(txn *badger.Txn) error {
+		// query for key/value
+		item, err := txn.Get([]byte(key))
+		if err != nil {
+			return fmt.Errorf("cannot get signature with key \"%s\": %w", key, err)
+		}
+
+		// copy value
+		err = item.Value(func(val []byte) error {
+			// Copying or parsing val is valid.
+			valCopy = append([]byte{}, val...)
+
+			return nil
+		})
+
+		return err
+	})
+
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return valCopy, nil
+}
+
+func validateKeyToken(s string) error {
+	if strings.Contains(s, ":") {
+		return errors.New("key token contains \":\"")
+	}
+
+	return nil
+}
+
+func getMetricKey(applicationName string, version int, signature, metric, user, transaction string) (string, error) {
+	if err := validateKeyToken(applicationName); err != nil {
+		return "", errors.New("application name cannot have \":\"")
+	}
+	if err := validateKeyToken(signature); err != nil {
+		return "", errors.New("signature cannot have \":\"")
+	}
+	if err := validateKeyToken(metric); err != nil {
+		return "", errors.New("metric name cannot have \":\"")
+	}
+	if err := validateKeyToken(user); err != nil {
+		return "", errors.New("user name cannot have \":\"")
+	}
+	if err := validateKeyToken(transaction); err != nil {
+		return "", errors.New("transaction ID cannot have \":\"")
+	}
+
+	return fmt.Sprintf("kt-metric::%s::%d::%s::%s::%s::%s", applicationName, version, signature, metric, user, transaction), nil
+}
+
+// GetMetric gets a metric based on the app name, version, signature, metric type, user, and transaction ID from BadgerDB
+// Example key/value: kt-metric::my-app::0::my-signature::my-metric::my-user::my-transaction-id -> my-metric-value
+func (cl Client) GetMetric(applicationName string, version int, signature, metric, user, transaction string) (float64, error) {
```

Comment:
Where is this used?

Reply:
It is being used in some test cases, I believe.

Reply:
Should I comment this out? The test, as well as `getValueFromBadgerDB()`?

Reply:
Yes, if a func is getting used only in test cases, we can comment it out. Can be reintroduced later if it is really needed.

Reply:
Done.